### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by EidHachem**
In this PR @Pazzo97 and I implemented the below:

- Added option selection method to `main.rb` file
- Created separate files for the following methods: `create person`, `create books`, and `create rentals`
- We made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- We followed SOLID principles.

### CODE REVIEW

File: methods/add_rentals.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+def add_rental
+  puts 'Select a book from the following list by number'
+  list_books(show_index: true)
+  book = gets.chomp.to_i
+  puts 'Invalid input!' if !(book.is_a? Integer) && book >= @books.length
+
+  puts 'Select a person from the following list by number (not id)'
+  list_people(show_index: true)
+  person = gets.chomp.to_i
+  puts 'Invalide input!' if !(person.is_a? Integer) && person >= @people.length
```
Comment (codepantha) [inline methods/add_rentals.rb:10]: - [x] Following the principle of single responsibility, please create a method whose sole responsibility is to request and validate input

File: methods/add_student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+def add_student
+  print 'Enter the name of the student: '
+  name = gets.chomp.to_s
+  print 'Enter the age of the student: '
+  age = gets.chomp.to_i
+  print 'Has parent permission? [Y/N]: '
```
Comment (codepantha) [inline methods/add_student.rb:6]: - [x] Following the principle of single responsibility, please create a method whose sole responsibility is to request and validate the student input

File: methods/add_teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+def add_teacher
+  print 'Enter the name of the teacher: '
+  name = gets.chomp.to_s
+  print 'Enter the age of the teacher: '
+  age = gets.chomp.to_i
+  print "And teacher's specialization: "
```
Comment (codepantha) [inline methods/add_teacher.rb:6]: - [x] Following the principle of single responsibility, please create a method whose sole responsibility is to request and validate teacher input

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️ 

## Hi @EidHachem and @Pazzo97 👋, 

You have done a good job so far 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

### Highlights ⭐ 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors
✔️ App working as expected

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi Team, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Clean and organized code.
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**